### PR TITLE
feat: squad category feed

### PR DESCRIPTION
**Pull request by sshanzel**
Note: the navbar will be applied once the PR for that is merged.

## Changes
- Squad category feed.
- Simplified the `UnfeaturedSquad` a bit based on its requirements.
- Updated the `SquadJoinButton` to have a variant of `ViewSquad` depending on how we want it to display.

Preview:
Screenshot 2024-09-06 at 8 14 15 PM

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

MI-487

### Preview domain
https://mi-487.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/cards/Card.tsx [hunk 1]
```diff
@@ -56,19 +56,19 @@ export const CardLink = classed('a', clickableCardClasses);
 export const Card = classed(
   'article',
   styles.card,
-  'snap-start relative max-h-cardLarge h-full flex flex-col p-2 rounded-16 bg-background-subtle border border-border-subtlest-tertiary hover:border-border-subtlest-secondary shadow-2',
```
Comment (sshanzel) [inline packages/shared/src/components/cards/Card.tsx:59]: Ignore the header*

Before:



After:
Screenshot 2024-09-06 at 7 46 45 PM
0QXQJNB/p1725538654411249

Figma:

Screenshot 2024-09-06 at 8 13 13 PM

Thankfully Tomer mentioned that the shadow should not actually be there generally.
https://dailydotdev.slack.com/archives/C01L

File: packages/shared/src/components/cards/Card.tsx [hunk 2]
```diff
@@ -56,19 +56,19 @@ export const CardLink = classed('a', clickableCardClasses);
...
   'article',
   styles.card,
-  'tablet:max-w-[21.5rem] w-full relative max-h-fit h-full flex flex-col bg-background-subtle tablet:border border-accent-cabbage-default hover:border-accent-cabbage-default shadow-2',
+  'tablet:max-w-[21.5rem] w-full relative max-h-fit h-full flex flex-col bg-background-subtle tablet:border border-accent-cabbage-default hover:border-accent-cabbage-default',
 );
 
 export const CardHeader = classed(
   'div',
   styles.header,
-  'flex items-center h-8 items-center my-1 mt-2 -mx-1.5',
```
Comment (rebelchris) [inline packages/shared/src/components/cards/Card.tsx:71]: This doesn't affect any other cards by chance?

File: packages/shared/src/components/cards/squad/UnfeaturedSquadGrid.tsx [hunk 3]
```diff
@@ -10,39 +10,44 @@ import {
...
         rel="noopener"
         title={source.description}
       />
       <div className="mb-3 flex items-center justify-between">
-        <SquadImage
-          image={source?.image}
-          icon={icon}
-          title={title}
-          size="size-16"
+        <img
```
Comment (rebelchris) [inline packages/shared/src/components/cards/squad/UnfeaturedSquadGrid.tsx:38]: Shouldn't we use `Image` here?

File: packages/webapp/pages/squads/discover/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+function SquadCategoryPage({ category }: SquadCategoryPageProps): ReactElement {
+  const { result } = useSources({
+    query: { categoryId: category.id, isPublic: true },
+  });
+  const flatSources =
+    result.data?.pages.flatMap((page) => page.sources.edges) ?? [];
+
+  return (
+    <div className="mx-auto mt-5 flex w-full max-w-[99.75rem] flex-col px-5">
```
Comment (rebelchris) [inline packages/webapp/pages/squads/discover/[id].tsx:30]: What's this max w 👀

**Review by rebelchris — APPROVED** [review]
Overall nothing blocking.

File: packages/shared/src/components/cards/squad/UnfeaturedSquadGrid.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/cards/squad/UnfeaturedSquadGrid.tsx:38]: Right. Yeah, let's use it wherever we can.

File: packages/webapp/pages/squads/discover/[id].tsx (hunk 4 above)
Comment (sshanzel) [reply packages/webapp/pages/squads/discover/[id].tsx:30]: Oh, I initially thought we would need some max-w, but looks like we actually don't. Should update this.

Screenshot 2024-09-10 at 12 02 29 AM

File: packages/shared/src/components/cards/Card.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/cards/Card.tsx:71]: Yep, just a duplicated code from this same line.

**Pull request merged**